libs/utils/decorate.js

```diff
@@ -0,0 +1,146 @@
+      i.classList.add(icon.key)
+      const svg = `<img height="${size}" width="${size}" alt="${icon.label}" src="${icon.value}">`;
+      const label = `${svg} ${(icon.label !== undefined) ? icon.label : ''}`;
+      const anchor = `<a class="icon ${str}" href="${icon.link}">${label}</a>`;
+      if (icon.link !== undefined) {
+        i.outerHTML = anchor;
+      } else {
+        i.innerHTML = i.innerHTML.replace(str, label);
+      }
+    }
+  });
+}
+
+export function decorateBlockText(el, size) {
+  const headings = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
+  const heading = headings[headings.length - 1];
+  if (!size || size === 'small') {
+    heading.classList.add('heading-XS');
+    heading.nextElementSibling.classList.add('body-S');
+    if (heading.previousElementSibling) {
+      heading.previousElementSibling.classList.add('detail-M');
+    }
+  }
+  if (size === 'medium') {
+    heading.classList.add('heading-M');
+    heading.nextElementSibling.classList.add('body-S');
+    if (heading.previousElementSibling) {
+      heading.previousElementSibling.classList.add('detail-M');
+    }
+  }
+  if (size === 'large') {
+    heading.classList.add('heading-XL');
+    heading.nextElementSibling.classList.add('body-M');
+    if (heading.previousElementSibling) {
+      heading.previousElementSibling.classList.add('detail-L');
+    }
+  }
+  initIcons(el);
+  decorateButtons(el);
+  decorateLinkAnalytics(el, heading);
+}
+
+export function isHexColorDark(color) {
+  if (color[0] !== '#') return false;
+  const hex = color.replace('#', '');
+  const cR = parseInt(hex.substr(0, 2), 16);
+  const cG = parseInt(hex.substr(2, 2), 16);
+  const cB = parseInt(hex.substr(4, 2), 16);
+  const brightness = ((cR * 299) + (cG * 587) + (cB * 114)) / 1000;
+  return brightness < 155;
+}
+
+export function decorateBlockBg(block, node) {
+  node.classList.add('background');
+  if (!node.querySelector(':scope img')) {
+    block.style.background = node.textContent;
+    if (isHexColorDark(node.textContent)) block.classList.add('dark');
+    node.remove();
+  }
+}
+
+export function getBlockSize(el) {
+  const sizes = ['small', 'medium', 'large'];
+  return sizes.reduce((rdx, size) => {
+    if (el.classList.contains(size)) {
+      return size;
+    }
+    return rdx;
+  }, sizes[1]);
+}
+
+export async function getIconLibrary(path = '/docs/library/tokens.json') {
+  let library = {};
+  const url = (window.location.port === '2000') ? `https://main--milo--adobecom.hlx.page${path}` : path;
+  const resp = await fetch(url);
+  if (!resp.ok) return;
+  const json = await resp.json();
+  json['icons']?.data.forEach((item) => {
+    const itemValues = {};
+    Object.entries(item).forEach((value) => {
+      const itemValue = value[1];
+      if (itemValue) {
+        itemValues[value[0]] = itemValue;
+      }
+    });
+    library[item.key] = itemValues;
+  });
+  await decorateIcons(library);
+}
+
+export async function loadTokens(blocks) {
```

Comment:
Should be moved into /libs/utils/utils.js, with `ICON_BLOCKS` defined at the top